<h1>Palmer Penguin Data Analytics</h1>

<H2>Analysis: Carla Mae M. Biñas</H2>

<p>Always restart the cells when you open the file!!!</p>

<h4>Importing numpy and pandas</h4>

In [69]:
import numpy as np 
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
from scipy.stats import f_oneway

<h4>Importing Datasets to Jupyter Notebook</h4>
<h4>Converting Datasets to Daataframe</h4>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

<p>Checking if the datasets successfully imported</p>

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<p>The use of .describe()</p>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<p>The use of .info()</p>

In [5]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


<h4>Selecting Columns</h4>

In [6]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [7]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

<h4> The use of Logical Operator</h4>

In [8]:
(penguins[penguins['species'] == 'Chinstrap'])

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [9]:
(penguins[penguins['body_mass_g'] >= 4000])

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

<p>The use of .unique()</p>

In [10]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

<h4>Checking the DataFrame</h4>

In [11]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

<h4>Cleaning the DataFrame</h4>

In [12]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>How to Fill up null values</h2>

In [13]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<p>The use of .mean()</p>

In [14]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [15]:
penguins['bill_length_mm']=penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [16]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [17]:
penguins['bill_depth_mm']=penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [18]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [19]:
penguins['flipper_length_mm']=penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [20]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [21]:
penguins['body_mass_g']=penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

<p>The use of .mode()</p>

In [23]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [24]:
penguins['sex']=penguins['sex'].fillna(penguins['sex'].mode()[0])

In [25]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h4> Changing the data type of the column</h4>

In [26]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [27]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

<p>The use of .astype()</p>

In [28]:
penguins['species']=penguins['species'].astype('category')

In [29]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(1), float64(4), int64(1), object(2)
memory usage: 19.4+ KB


In [30]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [31]:
penguins['island']=penguins['island'].astype('category')

In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(2), float64(4), int64(1), object(1)
memory usage: 17.2+ KB


In [33]:
penguins['sex']

0        male
1      female
2      female
3        male
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [34]:
penguins['sex']=penguins['sex'].astype('category')

In [35]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


In [36]:
penguins['year']

0      2007
1      2007
2      2007
3      2007
4      2007
       ... 
339    2009
340    2009
341    2009
342    2009
343    2009
Name: year, Length: 344, dtype: int64

In [37]:
penguins['year']=penguins['year'].astype('category')

In [38]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h4> Creating a new Column</h4>

In [39]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<p>The use of .columns()</p>

In [40]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [41]:
penguins['mass_kg']=np.round(penguins['body_mass_g']/1000,1)

In [42]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [43]:
penguins['mass_kg']

0      3.8
1      3.8
2      3.2
3      4.2
4      3.4
      ... 
339    4.0
340    3.4
341    3.8
342    4.1
343    3.8
Name: mass_kg, Length: 344, dtype: float64

<h2>The Exploratory Data Analytics (EDA)</h2>

<hr>

<h4>1. Which island has the most penguins in the dataset?</h4>

In [44]:
penguins['island'].mode()

0    Biscoe
Name: island, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

<h5>Insights: Therefore the most penguins in dataset came from Biscoe Island.</h5>

<hr>

<h5>Double check</h5>

In [45]:
(penguins[penguins['island'] == 'Biscoe'])

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20   Adelie  Biscoe        37.80000       18.30000         174.000000   
21   Adelie  Biscoe        37.70000       18.70000         180.000000   
22   Adelie  Biscoe        35.90000       19.20000         189.000000   
23   Adelie  Biscoe        38.20000       18.10000         185.000000   
24   Adelie  Biscoe        38.80000       17.20000         180.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  mass_kg  
20   3400.000000  female  2007      3.4  
21   3600.000000    male  2007      3.6  
22   3800.000000  female  2007      3.8  
23   3950.000000    male  2007      4.0  
24   3800.000000    male  2007      3.8  
..           ...     ...   ...      ...  
271  4201.754386    male  2009      4.2  
272  4850.000000  female  2009      4.8  
273  5750.000000    male  2009      5.8  
274  5200.000000  female  2009      5.2  
275  5400.000000    male  2009      5.4  

[168 rows x 9 columns]

In [46]:
(penguins[penguins['island'] == 'Dream'])

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007      3.2  
31        3900.0    male  2007      3.9  
32        3300.0  female  2007      3.3  
33        3900.0    male  2007      3.9  
34        3325.0  female  2007      3.3  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[124 rows x 9 columns]

In [47]:
(penguins[penguins['island'] == 'Torgersen'])

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000       18.70000         181.000000   
1    Adelie  Torgersen        39.50000       17.40000         186.000000   
2    Adelie  Torgersen        40.30000       18.00000         195.000000   
3    Adelie  Torgersen        43.92193       17.15117         200.915205   
4    Adelie  Torgersen        36.70000       19.30000         193.000000   
5    Adelie  Torgersen        39.30000       20.60000         190.000000   
6    Adelie  Torgersen        38.90000       17.80000         181.000000   
7    Adelie  Torgersen        39.20000       19.60000         195.000000   
8    Adelie  Torgersen        34.10000       18.10000         193.000000   
9    Adelie  Torgersen        42.00000       20.20000         190.000000   
10   Adelie  Torgersen        37.80000       17.10000         186.000000   
11   Adelie  Torgersen        37.80000       17.30000         180.000000   
12   Adelie  Torgersen        41.10000       17.60000         182.000000   
13   Adelie  Torgersen        38.60000       21.20000         191.000000   
14   Adelie  Torgersen        34.60000       21.10000         198.000000   
15   Adelie  Torgersen        36.60000       17.80000         185.000000   
16   Adelie  Torgersen        38.70000       19.00000         195.000000   
17   Adelie  Torgersen        42.50000       20.70000         197.000000   
18   Adelie  Torgersen        34.40000       18.40000         184.000000   
19   Adelie  Torgersen        46.00000       21.50000         194.000000   
68   Adelie  Torgersen        35.90000       16.60000         190.000000   
69   Adelie  Torgersen        41.80000       19.40000         198.000000   
70   Adelie  Torgersen        33.50000       19.00000         190.000000   
71   Adelie  Torgersen        39.70000       18.40000         190.000000   
72   Adelie  Torgersen        39.60000       17.20000         196.000000   
73   Adelie  Torgersen        45.80000       18.90000         197.000000   
74   Adelie  Torgersen        35.50000       17.50000         190.000000   
75   Adelie  Torgersen        42.80000       18.50000         195.000000   
76   Adelie  Torgersen        40.90000       16.80000         191.000000   
77   Adelie  Torgersen        37.20000       19.40000         184.000000   
78   Adelie  Torgersen        36.20000       16.10000         187.000000   
79   Adelie  Torgersen        42.10000       19.10000         195.000000   
80   Adelie  Torgersen        34.60000       17.20000         189.000000   
81   Adelie  Torgersen        42.90000       17.60000         196.000000   
82   Adelie  Torgersen        36.70000       18.80000         187.000000   
83   Adelie  Torgersen        35.10000       19.40000         193.000000   
116  Adelie  Torgersen        38.60000       17.00000         188.000000   
117  Adelie  Torgersen        37.30000       20.50000         199.000000   
118  Adelie  Torgersen        35.70000       17.00000         189.000000   
119  Adelie  Torgersen        41.10000       18.60000         189.000000   
120  Adelie  Torgersen        36.20000       17.20000         187.000000   
121  Adelie  Torgersen        37.70000       19.80000         198.000000   
122  Adelie  Torgersen        40.20000       17.00000         176.000000   
123  Adelie  Torgersen        41.40000       18.50000         202.000000   
124  Adelie  Torgersen        35.20000       15.90000         186.000000   
125  Adelie  Torgersen        40.60000       19.00000         199.000000   
126  Adelie  Torgersen        38.80000       17.60000         191.000000   
127  Adelie  Torgersen        41.50000       18.30000         195.000000   
128  Adelie  Torgersen        39.00000       17.10000         191.000000   
129  Adelie  Torgersen        44.10000       18.00000         210.000000   
130  Adelie  Torgersen        38.50000       17.90000         190.000000   
131  Adelie  Torgersen        43.10000       19.2000

<hr>

<h4>2. "How many penguins have a bill depth of at least 21 mm, and what are their species?"</h4>

In [48]:
(penguins[penguins['bill_depth_mm'] >= 21])

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
13  Adelie  Torgersen            38.6           21.2              191.0   
14  Adelie  Torgersen            34.6           21.1              198.0   
19  Adelie  Torgersen            46.0           21.5              194.0   
35  Adelie      Dream            39.2           21.1              196.0   
49  Adelie      Dream            42.3           21.2              191.0   
61  Adelie     Biscoe            41.3           21.1              195.0   

    body_mass_g   sex  year  mass_kg  
13       3800.0  male  2007      3.8  
14       4400.0  male  2007      4.4  
19       4200.0  male  2007      4.2  
35       4150.0  male  2007      4.2  
49       4150.0  male  2007      4.2  
61       4400.0  male  2008      4.4

<h5>Insights: The penguins that has 21 mm bill depth is 6 and they are all Adelie species.</h5>

<hr>

<h4>3. What is the average bill depth for Adelie penguins?</h4>

In [49]:
adelie_mean = penguins[penguins['species'] == 'Adelie']["bill_depth_mm"].mean()

In [50]:
adelie_mean

np.float64(18.338494536780548)

<h5>Insights: The average bill depth of the Adelie Species is 18.34.</h5>

<hr>

<h4>4. What are the most common bill lengths for Chinstrap penguins in the dataset?</h4>

In [51]:
(penguins[penguins['species'] == 'Chinstrap'] ['bill_length_mm'].mode())

0    51.3
1    52.0
Name: bill_length_mm, dtype: float64

<h5>Insights: The most common bill lengths for chinstrap species is 51.3</h5>

<hr>

<h4>5. What is the average weight of male and female penguins? Who weights more? </h4>

In [52]:
(penguins[penguins['sex'] == 'female'] ['mass_kg'].mean())

np.float64(3.8612121212121213)

In [53]:
(penguins[penguins['sex'] == 'male'] ['mass_kg'].mean())

np.float64(4.516201117318436)

<h5>Insights: The average weight of female penguins is 3.86 while the male penguins is 4.52. Since the male average is higher than the female, it clearly shows that male penguins generally weigh more than females.</h5>

<hr>

<h4>6. Which penguin species has the longest flipper length?</h4>

In [54]:
longest_flipper = np.max(penguins["flipper_length_mm"])

In [55]:
longest_flipper

np.float64(231.0)

<h5>Insights: The Longest flipper length is 231.</h5>

<hr>

<h4>7. Which island do penguins have the heaviest body weight?</h4>

In [56]:
heaviest_weight = np.max(penguins["mass_kg"])

In [57]:
h_island = penguins[penguins["mass_kg"] == heaviest_weight]["island"].values[0]

In [58]:
heaviest_weight, h_island

(np.float64(6.3), 'Biscoe')

<h5>Insights: The heaviest weight of penguin is 6.3 in Biscoe island.</h5>

<hr>

<h2>Data Aggregation</h2>

<hr>

In [59]:
PerIsland = penguins.groupby(['species', 'island', 'sex', 'year']).size()

In [60]:
PerIsland

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Biscoe     female  2007     0
                              2008     0
                              2009     0
                      male    2007     0
                              2008     0
                      

In [63]:
mass_species= penguins.groupby(['species'])['body_mass_g'].mean()

In [64]:
mass_species

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h2>Testing Correlation</h2>

<h1>Pearson R Correlation</h1>

In [65]:
weight_flipper_cor= penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [66]:
weight_flipper_cor

np.float64(0.8712017673060114)

<h2>Five Question using Correlation</h2>

<hr>

<h3>8. Does bill length increase with body mass?</h3>

In [83]:
penguins["bill_length_mm"].corr(penguins["body_mass_g"])

np.float64(0.5951098244376304)

<h4>Insight: There is a correlation between bill length and body mass. This means penguins with larger body mass usually have longer bills.</h4>

<hr>

<h3>9. Do penguins with longer flippers also tend to have longer bills?</h3>

In [85]:
penguins["flipper_length_mm"].corr(penguins["bill_length_mm"])

np.float64(0.6561813407464278)

<h4>Insight: There is a correlation between flipper length and bill length. Penguins with longer flippers also tend to have longer bills.</h4>

<hr>

<h3>10. Do heavier penguins tend to have longer bills?</h3>

In [86]:
penguins["body_mass_g"].corr(penguins["bill_length_mm"])

np.float64(0.5951098244376305)

<h4>Insight: Since the correlation is greater than 0.5, there is a relationship between body mass and bill length. It means heavier penguins tend to have longer bills.</h4>

<hr>

<h3>11. Is bill length related to bill depth?</h3>

In [87]:
penguins["bill_length_mm"].corr(penguins["bill_depth_mm"])

np.float64(-0.23505287035553282)

<h4>Insight: The correlation is less than 0.5, so we can say that bill length and bill depth are not related.</h4>

<hr>

<h3>12. Among different species, does body mass correlate with flipper length?</h3>

In [88]:
penguins.groupby("species")[["body_mass_g", "flipper_length_mm"]].corr()

body_mass_g  flipper_length_mm
species                                                    
Adelie    body_mass_g           1.000000           0.474054
          flipper_length_mm     0.474054           1.000000
Chinstrap body_mass_g           1.000000           0.641559
          flipper_length_mm     0.641559           1.000000
Gentoo    body_mass_g           1.000000           0.711209
          flipper_length_mm     0.711209           1.000000

<h4>Insight: Body mass and flipper length are positively correlated. Gentoo penguins show the strongest relationship, meaning body size and flipper size grow together more consistently compared to Adelie and Chinstrap.</h4>

<hr>

<h2>Library SciPy(Scientific Python)</h2>

<h5>Numerical - Categorical Correlational Test</h5>

In [70]:
penguins[penguins['sex']== 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [71]:
penguins[penguins['sex']== 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [72]:
x_df= penguins[penguins['sex']== 'male']

In [73]:
x_df['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [80]:
y_df= penguins[penguins['sex']== 'female']

In [82]:
y_df['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [76]:
result,pvalue= f_oneway(penguins[penguins['sex']=='male']['body_mass_g'], penguins[penguins['sex']=='female']['body_mass_g'])

In [78]:
pvalue

np.float64(3.0226600487050246e-15)

<h2>Extra Question Using Numerical and Categorical Correlational </h2>

<hr>

In [90]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: category
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [91]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h4>13. Which penguin species has the longest average bill length?</h4>

In [95]:
result,pvalue= f_oneway(penguins[penguins['species']=='Adelie']['bill_length_mm'], penguins[penguins['species']=='Chinstrap']['bill_length_mm'], penguins[penguins['species']=='Gentoo']['bill_length_mm'])

In [96]:
pvalue 

np.float64(7.233266188149441e-91)

<h6>I dont know how to show the pvalue of each species not unless I do it one by one</h6>